File: src/main/java/core/db/TransferTypeTable.java
package core.db;

import module.transfer.TransferType;

import java.sql.Types;

public class TransferTypeTable extends AbstractTable {

	final static String TABLENAME = "TRANSFERTYPE";
	
	TransferTypeTable(JDBCAdapter adapter){
		super(TABLENAME,adapter);
	}
	
	@Override
	protected void initColumns() {
        columns = new ColumnDescriptor[]{
                ColumnDescriptor.Builder.newInstance().setColumnName("PLAYER_ID").setGetter((p) -> ((TransferType) p).getPlayerId()).setSetter((p, v) -> ((TransferType) p).setPlayerId((int) v)).setType(Types.INTEGER).isPrimaryKey(true).isNullable(false).build(),
                ColumnDescriptor.Builder.newInstance().setColumnName("TYPE").setGetter((p) -> ((TransferType) p).getTransferType()).setSetter((p, v) -> ((TransferType) p).setTransferType((Integer) v)).setType(Types.INTEGER).isNullable(true).build()
        };
	}

    void storeTransferType(TransferType type) {
        type.setIsStored(isStored(type.getPlayerId()));
        store(type);
    }
    
    TransferType loadTransferType(int playerId) {
        return loadOne(TransferType.class, playerId);
    }

}

File: src/main/java/core/db/TransferTypeTable.kt
package core.db

import module.transfer.TransferType
import java.sql.*
import java.util.function.BiConsumer
import java.util.function.Function

class TransferTypeTable internal constructor(adapter: JDBCAdapter) : AbstractTable(TABLENAME, adapter) {
    override fun initColumns() {
        columns = arrayOf<ColumnDescriptor>(
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("PLAYER_ID")
                .setGetter(Function<Any?, Any?> { p: Any? -> (p as TransferType?)!!.playerId }).setSetter(
                BiConsumer<Any?, Any> { p: Any?, v: Any -> (p as TransferType?)!!.playerId = v as Int })
                .setType(Types.INTEGER).isPrimaryKey(true).isNullable(false).build(),
            ColumnDescriptor.Builder.Companion.newInstance().setColumnName("TYPE")
                .setGetter(Function<Any?, Any?> { p: Any? -> (p as TransferType?)!!.transferType }).setSetter(
                BiConsumer<Any?, Any> { p: Any?, v: Any? -> (p as TransferType?)!!.transferType = v as Int? })
                .setType(Types.INTEGER).isNullable(true).build()
        )
    }

    fun storeTransferType(type: TransferType) {
        type.stored = isStored(type.playerId)
        store(type)
    }

    fun loadTransferType(playerId: Int): TransferType? {
        return loadOne(TransferType::class.java, playerId)
    }

    companion object {
        const val TABLENAME = "TRANSFERTYPE"
    }
}
